src/hotspot/os/posix/os_posix.hpp

```diff
@@ -45,7 +45,7 @@
   } while(((int)_result == OS_ERR) && (errno == EINTR))
 
 #define RESTARTABLE_RETURN_INT(_cmd) do { \
-  int _result; \
+  ssize_t _result; \
```

Comment:
Ah no - please don't do this. Please define a new macro.

Reply:
Given that the three functions that use this return ssize_t, I've renamed the macro.